src/formatDate.js

```diff
@@ -8,7 +8,61 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat.find(
+    (item) => item === '/' || item === '-' || item === '.',
+  );
+
+  const separator = toFormat.find(
+    (item) => item === '/' || item === '-' || item === '.',
+  );
+
+  const newDate = date.split(fromSeparator);
+
+  const newArray = [];
+
+  const dateObject = {};
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    dateObject[fromFormat[i]] = newDate[i];
+  }
+
+  const indexOfYear = fromFormat.includes('YYYY')
+    ? fromFormat.indexOf('YYYY')
+    : fromFormat.indexOf('YY');
+
+  if (!('YYYY' in dateObject)) {
+    if (newDate[indexOfYear] >= 30) {
+      dateObject.YYYY = '19' + newDate[indexOfYear];
+    }
+
+    if (newDate[indexOfYear] < 30) {
+      dateObject.YYYY = '20' + newDate[indexOfYear];
+    }
+  }
+
+  if ('YYYY' in dateObject) {
+    dateObject.YY = newDate[indexOfYear].slice(2);
```

Comment:
The logic here assumes that `newDate[indexOfYear]` is always a string with at least two characters. Ensure that this assumption holds true or add checks to handle cases where it might not be valid.